test/resources/mnist/distributed_customer_script_with_env_vars.py

```diff
@@ -162,12 +163,8 @@ def _preprocess_mnist(raw, withlabel, ndim, scale, image_dtype, label_dtype, rgb
 
     trainer.run()
 
-    # the MPI command makes the SM_HOSTS variable an invalid JSON
-    hosts = os.environ['SM_HOSTS']
-    master_node = hosts[1:hosts.find(',')]
-
     # only save the model in the master node
-    if args.host == master_node:
+    if args.host == json.loads(os.environ['SM_HOSTS'])[0]:
```

Comment:
I would just put hosts in argparser as well. YMMV

Reply:
will do in a separate PR since this one's getting lengthy